lib/cocoapods-core/lockfile.rb

```diff
@@ -440,6 +453,13 @@ def generate_dependencies_data(podfile)
         podfile.dependencies.map(&:to_s).sort
       end
 
+      def generate_spec_repos(spec_repos)
```

Comment:
nit if you care: add doc for this method

Reply:
should this filter out sources with no specs? probably